test/lib/projectPreprocessor.js

```diff
@@ -1770,3 +1776,147 @@ test("isBeingProcessed: Deduped project is being ignored", (t) => {
 	const res = preprocessor.isBeingProcessed(parent, project);
 	t.deepEqual(res, true, "Project is being ignored");
 });
+
+
+test.serial("applyType", async (t) => {
+	const formatStub = sinon.stub();
+	const getTypeStub = sinon.stub(require("@ui5/builder").types.typeRepository, "getType")
+		.returns({
+			format: formatStub
+		});
+
+	const project = {
+		type: "pony",
+		metadata: {}
+	};
+
+	const preprocessor = new projectPreprocessor._ProjectPreprocessor({});
+	await preprocessor.applyType(project);
+
+	t.is(getTypeStub.callCount, 1, "getType got called once");
+	t.deepEqual(getTypeStub.getCall(0).args[0], "pony", "getType got called with correct type");
+
+	t.is(formatStub.callCount, 1, "format got called once");
+	t.is(formatStub.getCall(0).args[0], project, "format got called with correct project");
+});
+
+test.serial("checkProjectMetadata: Warning logged for deprecated dependencies", async (t) => {
+	const log = require("@ui5/logger");
+	const loggerInstance = log.getLogger("pony");
+	mock("@ui5/logger", {
+		getLogger: () => loggerInstance
+	});
+	const logWarnSpy = sinon.spy(loggerInstance, "warn");
+
+	// Re-require tested module
+	const projectPreprocessor = mock.reRequire("../../lib/projectPreprocessor");
+
+	const preprocessor = new projectPreprocessor._ProjectPreprocessor({});
+
+	const project1 = {
+		metadata: {
+			name: "root.project",
+			deprecated: true
+		}
+	};
+
+	// no warning should be logged for root level project
+	await preprocessor.checkProjectMetadata(null, project1);
+
+	const project2 = {
+		_level: 1,
+		metadata: {
+			name: "my.project",
+			deprecated: true
+		}
+	};
+
+	// one warning should be logged for deprecated dependency
+	await preprocessor.checkProjectMetadata(project1, project2);
+
+	t.is(logWarnSpy.callCount, 1, "One warning got logged");
+	t.deepEqual(logWarnSpy.getCall(0).args[0],
+		"Dependency my.project is deprecated and should not be used for new projects!",
+		"Logged expected warning message");
+	mock.stop("@ui5/logger");
```

Comment:
You might also use `mock.stopAll()` within `test.afterEach.always` to ensure a proper cleanup

Reply:
Done